core/lib/types/src/api/mod.rs

```diff
@@ -736,3 +737,38 @@ pub struct Proof {
     pub address: Address,
     pub storage_proof: Vec<StorageProof>,
 }
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct OptimisticTransactionResult {
+    pub transaction_hash: H256,
+    pub storage_logs: Vec<ApiStorageLogQuery>,
+    pub events: Vec<ApiVmEvent>,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ApiVmEvent {
+    pub l1_batch_number: L1BatchNumber,
+    pub tx_batch_index: u32,
+    pub address: Address,
+    pub indexed_topics: Vec<H256>,
+    pub value: String,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ApiStorageLogQuery {
+    pub timestamp: Timestamp,
+    pub tx_number_in_block: u16,
+    pub aux_byte: u8,
+    pub shard_id: u8,
+    pub address: Address,
+    pub key: U256,
+    pub read_value: U256,
+    pub written_value: U256,
+    pub rw_flag: bool,
+    pub rollback: bool,
+    pub is_service: bool,
+    pub log_type: StorageLogQueryType,
+}
```

Comment:
Most of these fields are strictly internal, with some being only useful for proof generation. What lens is really interested in is only the updates to storage, which is `(address, key) => new_value` mapping.

Reply:
Fixed - filtering out all read only logs as discussed